nuxeo-runtime/nuxeo-runtime-stream/src/main/java/org/nuxeo/runtime/avro/AvroFactoryServiceImpl.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     pierre
+ */
+package org.nuxeo.runtime.avro;
+
+import java.lang.reflect.Constructor;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+/**
+ * @since 10.2
+ */
+public class AvroFactoryServiceImpl implements AvroFactoryService {
+
+    private final Map<Class<?>, Class<AvroFactory<?>>> registry = new HashMap<>(8);
+
+    private final SortedSet<AvroReplacementDescriptor> replacements = new TreeSet<>();
+
+    @Override
+    public AvroFactoryContext createContext() {
+        AvroFactoryContext context = new AvroFactoryContext(replacements);
+        for (Entry<Class<?>, Class<AvroFactory<?>>> entry : registry.entrySet()) {
+            Constructor<AvroFactory<?>> factoryConstructor;
+            try {
+                factoryConstructor = entry.getValue().getConstructor(AvroFactoryContext.class);
+                AvroFactory<?> factory = factoryConstructor.newInstance(context);
```

Comment:
`newInstance` is deprecated in Java 9. Please use `getConstructor(AvroFactoryContext.class).newInstance(context)`